packages/automated_actions_utils/automated_actions_utils/aws_api.py

```diff
@@ -0,0 +1,138 @@
+        region: The AWS region for the credentials.
+
+    Returns:
+        An AWSStaticCredentials object populated with credentials from Vault.
+
+    Raises:
+        SecretFieldNotFoundError: If 'aws_access_key_id' or 'aws_secret_access_key'
+                                  are not found in the secret.
+    """
+    vault_client = VaultClient(
+        server_url=settings.vault_server_url,
+        role_id=settings.vault_role_id,
+        secret_id=settings.vault_secret_id,
+        kube_auth_role=settings.vault_kube_auth_role,
+        kube_auth_mount=settings.vault_kube_auth_mount,
+    )
+
+    secret = vault_client.read_secret(
+        path=vault_secret.path,
+        version=str(vault_secret.version),
+    )
+
+    if "aws_access_key_id" not in secret or "aws_secret_access_key" not in secret:
+        raise SecretFieldNotFoundError(
+            f"aws_access_key_id or aws_secret_access_key not found in secret {vault_secret.path}"
+        )
+
+    return AWSStaticCredentials(
+        access_key_id=secret["aws_access_key_id"],
+        secret_access_key=secret["aws_secret_access_key"],
+        region=region,
+    )
+
+
+class AWSApi:
+    """A client for interacting with AWS services."""
+
+    def __init__(self, credentials: AWSCredentials, region: str | None) -> None:
+        self.session = credentials.build_session()
+        self.config = Config(
+            region_name=region,
+            retries={
+                "max_attempts": 5,
+                "mode": "standard",
+            },
+        )
+
+    @property
+    def rds_client(self) -> "RDSClient":
+        """Gets a boto3 RDS client."""
+        return self.session.client("rds", config=self.config)
+
+    def reboot_rds_instance(self, identifier: str, *, force_failover: bool) -> None:
+        """Reboots a specified RDS database instance.
+
+        Args:
+            identifier: The DB instance identifier.
+            force_failover: When true, the reboot is conducted through a MultiAZ failover.
+                            Cannot be set to true if the instance is not configured for MultiAZ.
+        """
+        log.info(f"Rebooting RDS instance {identifier}")
+        self.rds_client.reboot_db_instance(
+            DBInstanceIdentifier=identifier, ForceFailover=force_failover
+        )
+
+    def rds_get_events(
+        self,
+        identifier: str,
+        duration_min: int = 60,
+    ) -> list["EventTypeDef"]:
+        """Retrieves events for a specified RDS instance.
+
+        Args:
+            identifier: The DB instance identifier.
+            duration_min: The duration in minutes for which to retrieve events.
+
+        Returns:
+            A list of events.
+        """
+        log.info(f"Retrieving RDS events for {identifier}")
+        response = self.rds_client.describe_events(
+            SourceIdentifier=identifier,
+            SourceType="db-instance",
+            Duration=duration_min,
+        )
```

Comment:
this result can have multiple pages, require paginator https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html

Reply:
Changed in [516b0fc](https://github.com/app-sre/automated-actions/pull/95/commits/516b0fc55eb2190be9eb77e9336746b8290345e6)